Adjust unnecessarily precise description of superfluous-parens message

### Bug description

My initial code was:
```
if (i - 0) in exclude:
    continue
````
Which trigger superfluous-parens. I think the parentheses are technically unnecessary but it doesn't seem match with the intended target of this warning:
```
if (x == y):
    pass
```
I adjusted my code to:
```
foo = True
if foo and (i - 0) in exclude:
    continue
``` 
Which does not trigger superfluous-parens.
This makes me think that the warning is reacting to the proximity of the `if` and the `(` not the context in which they are being used.

### Configuration

_No response_

### Command used

```shell
python3 -m pylint --exit-zero --rcfile=pylint-config.txt scripts/tamper.py
```


### Pylint output

```shell
scripts/tamper.py:152:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
```


### Expected behavior

No superfluous-parens warning.

### Pylint version

```shell
pylint 2.12.2
```


### OS / Environment

macOS Monterey 12.2.1

### Additional dependencies

_No response_

--- a/doc/data/messages/s/superfluous-parens/bad.py
+++ b/doc/data/messages/s/superfluous-parens/bad.py
@@ -1,4 +1,4 @@
 x = input()
 y = input()
-if x == y:
+if (x == y):
     pass

--- a/doc/data/messages/s/superfluous-parens/bad.py
+++ b/doc/data/messages/s/superfluous-parens/bad.py
@@ -1,9 +1,9 @@
 x = input()
 y = input()
-if (x == y):
+if (x == y):  # [superfluous-parens]
     pass
 
 i = 0
 exclude = []
-if (i - 0) in exclude:
+if (i - 0) in exclude:  # [superfluous-parens]
     pass

--- a/doc/data/messages/s/superfluous-parens/bad.py
+++ b/doc/data/messages/s/superfluous-parens/bad.py
@@ -2,3 +2,8 @@ x = input()
 y = input()
 if (x == y):
     pass
+
+i = 0
+exclude = []
+if (i - 0) in exclude:
+    pass

--- a/doc/data/messages/s/superfluous-parens/details.rst
+++ b/doc/data/messages/s/superfluous-parens/details.rst
@@ -1,1 +1,0 @@
-You can help us make the doc better `by contributing <https://github.com/PyCQA/pylint/issues/5953>`_ !

--- a/doc/data/messages/s/superfluous-parens/good.py
+++ b/doc/data/messages/s/superfluous-parens/good.py
@@ -2,3 +2,8 @@ x = input()
 y = input()
 if x == y:
     pass
+
+i = 0
+exclude = []
+if i - 0 in exclude:
+    pass

